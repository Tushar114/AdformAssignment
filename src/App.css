.App {
  text-align: center;
}
.flex-container {
  display: flex;
}
.search-right {
  margin-left: auto;
}
