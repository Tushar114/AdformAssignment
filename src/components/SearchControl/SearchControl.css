.search-right {
  margin-left: auto;
}
