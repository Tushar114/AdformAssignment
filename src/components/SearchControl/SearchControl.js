import React from 'react';

function SearchControl({ handleSearch, searchText }) {
  return (
    <div className="search-right">
      <input
        type="text"
        value={searchText}
        placeholder="Search by name"
        onChange={handleSearch}
      ></input>
    </div>
  );
}

export default SearchControl;
