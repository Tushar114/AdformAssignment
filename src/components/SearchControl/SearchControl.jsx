import React from 'react';
import './SearchControl.css';
function SearchControl({ handleSearch, searchText }) {
  return (
    <div className="search-right">
      <input
        type="text"
        value={searchText}
        placeholder="Search by name"
        onChange={handleSearch}
      ></input>
    </div>
  );
}

export default SearchControl;
