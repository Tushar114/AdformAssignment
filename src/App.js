import React, { Component } from 'react';
import './App.css';
import DatePicker from 'react-datepicker';
import { connect } from 'react-redux';
import { fetch, dateFilter } from './actions';
import Loader from './Loader';
import Camptable from './Camptable';

class App extends Component {
  state = {
    startDate: '',
    endDate: '',
    searchText: '',
    isDateChanged: false,
  };

  componentDidMount() {
    this.props.fetch();
  }
  handleStartChange = (date) => {
    this.setState({
      startDate: date,
    });
  };
  handleEndChange = (date) => {
    this.setState(
      {
        endDate: date,
        isDateChanged: true,
      },
      () => {
        this.props.dateFilter(this.state.startDate, this.state.endDate);
      }
    );
  };
  handleSearch = (e) => {
    this.setState({ searchText: e.target.value });
  };
  render() {
    return (
      <div className="container">
        {this.props.isLoading ? (
          <Loader />
        ) : (
          <React.Fragment>
            <div className="mt-4">
              <DatePicker
                placeholderText="Start Date"
                selected={this.state.startDate}
                showMonthDropdown
                showYearDropdown
                onChange={this.handleStartChange}
              />
              <DatePicker
                placeholderText="End Date"
                selected={this.state.endDate}
                showMonthDropdown
                showYearDropdown
                onChange={this.handleEndChange}
              />
              <div className="search-right">
                <input
                  type="text"
                  value={this.state.searchText}
                  placeholder="Search by name"
                  onChange={this.handleSearch}
                ></input>
              </div>
              <div className="mt-4">
                {this.props.userData && (
                  <Camptable
                    userData={this.props.userData}
                    searchText={this.state.searchText}
                    startDate={this.state.startDate}
                    endDate={this.state.endDate}
                    isDateChanged={this.state.isDateChanged}
                  />
                )}
              </div>
            </div>
          </React.Fragment>
        )}
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    isLoading: state.campaign.loading,
    userData: state.campaign.data,
  };
};
export default connect(mapStateToProps, { fetch, dateFilter })(App);
