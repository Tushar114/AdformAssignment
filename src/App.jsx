import React, { useState, useEffect } from 'react';
import './App.css';
import { useSelector, useDispatch } from 'react-redux';
import { fetchUser } from './middleware/Middleware';
import Loader from './components/Loader/Loader.jsx';
import Camptable from './components/Camptable.jsx';
import { SelectUserData, SelectIsLoading, SelectError } from './selector';
import DateSelector from './components/DateSelector';
import SearchCamp from './components/SearchCamp';
import Error from './components/Error';

function App() {
  const [searchText, setSearchText] = useState('');
  const [isDateChanged, setIsDateChanged] = useState(false);
  const isLoading = useSelector((state) => SelectIsLoading(state));
  const userData = useSelector((state) => SelectUserData(state));
  const error = useSelector((state) => SelectError(state));

  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchUser());
  }, []);

  const handleSearch = (e) => {
    setSearchText(e.target.value);
  };

  return (
    <div className="container">
      {isLoading ? (
        <Loader />
      ) : (
        <React.Fragment>
          <div className="mt-4">
            {error ? (
              <Error />
            ) : (
              <>
                <div className="flex-container">
                  <DateSelector setIsDateChanged={setIsDateChanged} />
                  <SearchCamp
                    handleSearch={(e) => handleSearch(e)}
                    searchText={searchText}
                  />
                </div>
                <div className="mt-4">
                  {userData && (
                    <Camptable
                      searchText={searchText}
                      isDateChanged={isDateChanged}
                    />
                  )}
                </div>
              </>
            )}
          </div>
        </React.Fragment>
      )}
    </div>
  );
}

export default App;
