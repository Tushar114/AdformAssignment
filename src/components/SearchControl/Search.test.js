import React from 'react';
import App from '../../App.jsx';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import store from '../../redux/store';
import renderer from 'react-test-renderer';
import SearchControl from './SearchControl';

describe('Search', () => {
  it('should match the snapshot', () => {
    const component = renderer.create(<SearchControl />).toJSON();
    expect(component).toMatchSnapshot();
  });
  it('should render search component', () => {
    const component = shallow(<SearchControl />);
    const wrapper = component.find('.search-right');
    expect(wrapper.length).toBe(1);
  });

  it('search text is echoed', () => {
    const wrapper = shallow(<SearchControl handleSearch={() => {}} />);
    const app = mount(
      <Provider store={store}>
        <App />
      </Provider>
    );

    wrapper.find('input').simulate('change', {
      target: { value: 'Campaign 2' },
    });
    expect(app.searchText).toEqual('Campaign 2');
    //expect(wrapper.find('input').props().value).toEqual('Campaign 2');
  });

  it('checks if right search text is shared', () => {
    const wrapper = shallow(<SearchControl searchText="Campaign 2" />);
    const searchValue = wrapper.find('input').props().value;
    expect(searchValue).toEqual('Campaign 2');
  });
});
