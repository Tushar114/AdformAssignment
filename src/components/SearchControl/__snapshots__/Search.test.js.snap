// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match the snapshot 1`] = `
<div
  className="search-right"
>
  <input
    placeholder="Search by name"
    type="text"
  />
</div>
`;
